### PR TITLE
[KOGITO-7705 Updating grpc documentation

### PR DESCRIPTION
**Pull request by fjtirado**
Including properties and describing enum default behaviour

**JIRA:**
https://issues.redhat.com/browse/KOGITO-7705

**Description:**

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] The nav.adoc file has a link to this guide in the proper category
- [ ] The index.adoc file has a card to this guide in the proper category, with a meaningful description

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

**Review by MarianMacik — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-grpc-services.adoc [hunk 1]
```diff
@@ -145,6 +146,11 @@ quarkus.grpc.clients.Greeter.port=50051
 
 In the same GitHub repository as the example application, there is a link:{grpc_example_url}/serverless-workflow-greeting-server-rpc-quarkus[Maven project] which provides a simple implementation of the `Greeter` service. By default, this gRPC server runs on port `50051`. Therefore, the same port is used in the `application.properties` file.
 
+=== Default enum values
+
+grpc link:{grpc_enum_url}[specification] requires enumeration types to have a 0 default value. Default value is not included in the server response payload. Therefore it is recommended to use as default a meaningless value like UNKNOWN.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-grpc-services.adoc:151]: what do you mean by have a 0 default value?
Isn't better just say that "requires enum types to have no default value" ?

Comment (fjtirado) [reply serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-grpc-services.adoc:151]: The fact that the default value 0 is not really relevant so I will remove that for clarity

**Comment by ricardozanini** [comment]
Can you make this snippet multiline?

Screen Shot 2022-08-17 at 12 49 08

**Comment by fjtirado** [comment]
> Can you make this snippet multiline?
> 
> Screen Shot 2022-08-17 at 12 49 08

Im not a fan of using a PR for a purpose was not created for. Also I believe this should be single line after all

**Comment by ricardozanini** [comment]
> > Can you make this snippet multiline?
> > Screen Shot 2022-08-17 at 12 49 08
> 
> @ricardozanini Im not a fan of using a PR for a purpose was not created for. Also I believe this should be single line after all (splitting it might cause confusion to users, it is one or multiple lines?)

it's a simple change since you are updating this guide, after all. Here you go:

```shell
curl -X POST -H 'Content-Type:application/json' -H 'Accept:application/json' \
  -d '{"workflowdata" : { \
          "helloRequests" : [ \
            {"name" : "Javierito", "language":"Spanish"}, \
            {"name" : "John", "language":"English"}, \
            {"name" : "Jan", "language":"Czech"} \
      ]}}' \
http://localhost:8080/jsongreetclientstream
```

File: serverlessworkflow/modules/ROOT/pages/core/configuration-properties.adoc [hunk 2]
```diff
@@ -99,6 +99,18 @@ a|Defines strategy to generate the configuration key of open API specifications.
 |`_swf_eval_temp`
 |No
 
+|`kogito.grpc.enum.includeDefault`
+|Indicates if default value of enumerations should be included in grpc response.
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/core/configuration-properties.adoc:103]:
```suggestion
|Indicates if the default value of enumerations needs to be included in gRPC response.
```

File: serverlessworkflow/modules/ROOT/pages/core/configuration-properties.adoc [hunk 3]
```diff
@@ -99,6 +99,18 @@ a|Defines strategy to generate the configuration key of open API specifications.
...
+|boolean
+|false
+|No
+
+|`kogito.grpc.stream.timeout`
+|Number of seconds to wait for response from server when using streams
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/core/configuration-properties.adoc:109]:
```suggestion
|Number of seconds to wait for a response from the server when using streams.
```

File: serverlessworkflow/modules/ROOT/pages/core/configuration-properties.adoc [hunk 4]
```diff
@@ -99,6 +99,18 @@ a|Defines strategy to generate the configuration key of open API specifications.
 |`_swf_eval_temp`
 |No
 
+|`kogito.grpc.enum.includeDefault`
+|Indicates if default value of enumerations should be included in grpc response.
+|boolean
+|false
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/core/configuration-properties.adoc:105]:
```suggestion
|`false`
```

File: serverlessworkflow/modules/ROOT/pages/core/configuration-properties.adoc [hunk 5]
```diff
@@ -99,6 +99,18 @@ a|Defines strategy to generate the configuration key of open API specifications.
...
+|No
+
+|`kogito.grpc.stream.timeout`
+|Number of seconds to wait for response from server when using streams
+|int
+|20
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/core/configuration-properties.adoc:111]:
```suggestion
|`20`
```

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-grpc-services.adoc [hunk 6]
```diff
@@ -145,6 +146,11 @@ quarkus.grpc.clients.Greeter.port=50051
 
 In the same GitHub repository as the example application, there is a link:{grpc_example_url}/serverless-workflow-greeting-server-rpc-quarkus[Maven project] which provides a simple implementation of the `Greeter` service. By default, this gRPC server runs on port `50051`. Therefore, the same port is used in the `application.properties` file.
 
+=== Default enum values
+
+grpc link:{grpc_enum_url}[specification] requires enumeration types to have a default value. Default value is not included in the server response payload. Therefore it is recommended to use as default a meaningless value like UNKNOWN.
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-grpc-services.adoc:151]:
```suggestion
gRPC link:{grpc_enum_url}[specification] requires enumeration types to have a default value. The default value is not included in the server response payload. Therefore, use an empty value such as `UNKNOWN` as default.
```

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-grpc-services.adoc [hunk 7]
```diff
@@ -145,6 +146,11 @@ quarkus.grpc.clients.Greeter.port=50051
 
 In the same GitHub repository as the example application, there is a link:{grpc_example_url}/serverless-workflow-greeting-server-rpc-quarkus[Maven project] which provides a simple implementation of the `Greeter` service. By default, this gRPC server runs on port `50051`. Therefore, the same port is used in the `application.properties` file.
 
+=== Default enum values
+
+grpc link:{grpc_enum_url}[specification] requires enumeration types to have a default value. Default value is not included in the server response payload. Therefore it is recommended to use as default a meaningless value like UNKNOWN. 
+If, for any reason, your default value is semantically valid and you want it to be included in the Workflow model, you must set `kogito.grpc.enum.includeDefault` to true. Doing so, enumeration fields are always filled by the Workflow if the server response does not include them.
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-grpc-services.adoc:152]:
```suggestion
If, for any reason, your default value is semantically valid and you want the value to be included in the workflow model, you must set `kogito.grpc.enum.includeDefault` property to true. This way enumeration fields are always filled by the workflow if the server response does not include them. 
```

**Review by hmanwani-rh — COMMENTED** [review]
@fjtirado Please see the following suggestions, and let me know if you've any questions.

**Pull request merged**